### PR TITLE
Implement `filter_mode` for both `select` and `multiselect` prompts

### PR DESCRIPTION
**Pull request by fadeevab**
1. List renders through the `self.filter` always.
2. `Rc<RefCell>` is to handle a subset of filtered mutable references to the items from the
same structure.

### CODE REVIEW

File: src/multiselect.rs [hunk 1]
```diff
@@ -131,21 +162,35 @@ impl<T: Clone> PromptInteraction<Vec<T>> for MultiSelect<T> {
...
+            }
+        } else {
+            "".to_string()
+        };
 
-        let mut line2 = String::new();
-        for (i, item) in self.items.iter().enumerate() {
-            line2.push_str(&theme.format_multiselect_item(
+        let mut items_render = String::new();
+        for (i, item) in self.filter.items().iter().map(|i| i.borrow()).enumerate() {
```
Comment (fadeevab) [inline src/multiselect.rs:177]: Then, just replacing `self.items` to `self.filter.items()`. The same in `Select`.

**Comment by cylewitruk** [comment]
Will give it a try tomorrow! 😄

**Comment by antoinePutmans** [comment]
Good job @fadeevab  💪 I will test it tomorrow as well 😄

**Comment by antoinePutmans** [comment]
@fadeevab Hey, I tested the new feature, it works fine 👍 
The only thing that I would say feels a bit weird when using the filter on the multiselect is that the selected items wont be sent if the filter does not match them.

https://github.com/fadeevab/cliclack/assets/113629846/1236bd3c-fad1-4a75-b1b7-5bbf96a8acc1

For example here, Prettier wont be in the result even if he's selected.

**Comment by fadeevab** [comment]
> @fadeevab Hey, I tested the new feature, it works fine 👍 The only thing that I would say feels a bit weird when using the filter on the multiselect is that the selected items wont be sent if the filter does not match them.

Interesting, I haven't thought about it... I think a similar problem is in the `select`: it doesn't allow to select something invisible, and also it doesn't preserve the cursor once the filter was cleared.

**Comment by fadeevab** [comment]
@antoinePutmans Okay, I did this: it holds the item selection for multi-select list even when they are filtered out
image

and after `Enter` it rendered the items that were actually selected:
image

**Comment by fadeevab** [comment]
Superseded by PR #52

**Pull request closed without merging**